From 2e3ca9e280f8b9a429a35dcf6407f41dc7c35a18 Mon Sep 17 00:00:00 2001
From: "Arnaud Patard (Rtp)" <arnaud.patard@rtp-net.org>
Date: Fri, 25 Sep 2020 20:36:57 +0200
Subject: [PATCH 3/9] RFC: drivers/video/rockchip/rk_edp.c: Change interrupt
 polarity configuration

The linux code is setting polarity configuration to 3 but
uboot code is setting it to 1. Change the configuration to match the
linux configuration

Signed-off-by: Arnaud Patard <arnaud.patard@rtp-net.org>
---
 drivers/video/rockchip/rk_edp.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_edp.c b/drivers/video/rockchip/rk_edp.c
index 1b2f5f706d..13fa78aced 100644
--- a/drivers/video/rockchip/rk_edp.c
+++ b/drivers/video/rockchip/rk_edp.c
@@ -100,10 +100,13 @@ static void rk_edp_init_refclk(struct rk3288_edp *regs, enum rockchip_dp_types c
 	       &regs->dp_reserv2);
 }
 
+#define INT_POL1                                (0x1 << 1)
+#define INT_POL0                                (0x1 << 0)
+
 static void rk_edp_init_interrupt(struct rk3288_edp *regs)
 {
 	/* Set interrupt pin assertion polarity as high */
-	writel(INT_POL, &regs->int_ctl);
+	writel(INT_POL0 | INT_POL1, &regs->int_ctl);
 
 	/* Clear pending registers */
 	writel(0xff, &regs->common_int_sta_1);
-- 
2.28.0

